File: gatormoisture.cpp
/**
* Cece Hedrick
* June 16, 2020
* 
*
* Development environment specifics:
* Written in Microsoft PXT
*
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/


#include "pxt.h"
#include <cstdint>
#include <math.h>

using namespace pxt;

namespace gatormoisture {
    /*
    * Calculates the moisture level in soil 
    * The full range of the soil sensor ADC is (1023)
    * The value returned is a number between 0 and 1. 
    * 0 is dry 1 is wet 
    */
    //%
    uint16_t getMoisture(int16_t ADCVal) {

        return ADCVal / 1023.0;
    }

}

File: gatormoisture.ts
/**
* Cece Hedrick
* June 16 2020
* 

* Development environment specifics:
* Written in Microsoft Makecode
*
* This code is released under the [MIT License](http://opensource.org/licenses/MIT).
* Please review the LICENSE.md file included with this example. If you have any questions
* or concerns with licensing, please contact techsupport@sparkfun.com.
* Distributed as-is; no warranty is given.
*/


/**
 * Functions to operate the gatormoisture sensor
 */

 enum gatorMoistureType{
	 moisture=1,
	 adcVal=2,
 }



//% color=#f44242 icon="\uf185"
namespace gatorMoisture {

    // Functions for reading moisture from the gatormoisture in moisture or straight adv value

    /**
    * Reads the number
    */
    //% weight=30 blockId="gatorMoisture_moisture" block="Get moisture on pin %pin | in %gatorMoistureType"
    export function moisture(pin: AnalogPin, type: gatorMoistureType): number{
      let ADCVal = pins.analogReadPin(pin)
      switch(type){
        case gatorMoistureType.moisture: return getMoisture(ADCVal)
        case gatorMoistureType.adcVal: return ADCVal
        default: return -11111111
      }
    }

	/**
     * Function used for simulator, actual implementation is in gatormoisture.cpp
     */
    //% shim=gatorMoisture::getMoisture
    function getMoisture(ADCVal: number) {
        // Fake function for simulator
        return 0
    }

    }
